### PR TITLE
hugo: 0.136.5 -> 0.138.0

### PR DESCRIPTION
**Pull request by ghost**
https://github.com/gohugoio/hugo/releases/tag/v0.137.0
https://github.com/gohugoio/hugo/releases/tag/v0.137.1
https://github.com/gohugoio/hugo/releases/tag/v0.138.0

I added `withdeploy` as to not break compatibility with previous versions, maybe putting it behind a flag would be a good idea.

For reference, from the v0.137.0 release notes:

> Note that we have no longer build the [deploy feature](https://gohugo.io/hosting-and-deployment/hugo-deploy/) in the standard and extended archives. If you need that, download archives with withdeploy in the filename. See [instalation](https://gohugo.io/installation/) for details. You can read https://github.com/gohugoio/hugo/issues/12994 for more on the motivation behind this, but for one, it shaves off about 40% of the binary size.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by FedericoSchonborn** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>hugo</li>
  </ul>
</details>

File: pkgs/by-name/hu/hugo/package.nix [hunk 1]
```diff
@@ -12,22 +12,25 @@
...
+  vendorHash = "sha256-5YS76L7kisyPz8yv2RCgZHpY/AkjdHE+SUwMOuo3uLg=";
 
   doCheck = false;
 
   proxyVendor = true;
 
-  tags = [ "extended" ];
+  tags = [
+    "extended"
+    "withdeploy"
```
Comment (drupol) [inline pkgs/by-name/hu/hugo/package.nix:32]: Why this change?

**Review by drupol — APPROVED** [review]
Diff LGTM

File: pkgs/by-name/hu/hugo/package.nix (hunk 1 above)
Comment (FedericoSchonborn) [reply pkgs/by-name/hu/hugo/package.nix:32]: Hugo v0.137 put `hugo deploy` behind that build tag, so I added it in case anyone is using that feature nowadays.
